projects/admin/src/app/record/detail-view/document-detail-view/holding/holding.component.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, OnInit, Input } from '@angular/core';
+import { RecordService } from '@rero/ng-core';
+
+@Component({
+  selector: 'admin-document-holding',
+  templateUrl: './holding.component.html',
+  styles: ['./holding.component.css']
+})
+export class HoldingComponent implements OnInit {
+
+  @Input()
+  holding: any;
+
+  @Input()
+  harvested: false;
+
+  /** Holding location record */
+  location: any;
+
+  /** Items */
+  items: any;
+
+  /** Items collapsed */
+  isItemsCollapsed = false;
+
+  constructor(
+    private recordService: RecordService
+  ) { }
+
+  ngOnInit() {
+    if (!this.harvested) {
+      this.loadItems(this.holding.metadata.pid);
+    }
+    this.recordService
+      .getRecord('locations', this.holding.metadata.location.pid)
+      .subscribe(location => this.location = location);
+  }
+
+  toggleCollapse() {
+    if (this.isItemsCollapsed) {
+      this.loadItems(this.holding.metadata.pid);
+    }
+    this.isItemsCollapsed = !this.isItemsCollapsed;
+  }
+
+  deleteItem(itemPid: Event) {
+    this.items = this.items.filter((item: any) => itemPid !== item.metadata.pid);
+  }
+
+  private loadItems(holdingPid: string) {
+    const query = `holding.pid:${holdingPid}`;
+    this.recordService
+      .getRecords('items', query, 1, 100)
```

Comment:
you can use MAX results variable instead of 100

Reply:
done